static/js/main.js

```diff
@@ -0,0 +1,168 @@
+// Due to the way processing works, some variables that are treated as constants can't be declared as such, because they must be set in `setup()`.
+// We could probably avert this issue by severing the processing library from this project, and rewriting it in terms of DOM interaction.
+
+import { timeFormatter } from "./presets.js";
+import { getTargetTime, getTime, getTimeDifference } from "./timeUtils.js";
+import TimeInformation from "./timeInformation.js";
+
+const modes = ["60x60", "25x25", "45x10", "1x1", "0.1x1", "0.02x0.02", "90x90"]; //add modes here
+
+var timerRunning = false;
+var buttons = [];
+var currentWorkTime, currentRestTime;
+var working = true; //false=resting
+var currentButtonsAmount = 0;
+var buttonWidth, buttonHeight; // Should be constant, but due to the way processing works, must be set in `setup()`.
+var endTime;
+var resetTimer = false; //for pomodoro screen
+var theText; //for pomodoro screen - global
+var timeLeft;
+var soundEffect; // Should be constant.
+
+function setup() {
+	buttonWidth = windowWidth / 5;
+	buttonHeight = windowHeight / 10;
+
+	createCanvas(windowWidth, windowHeight);
+
+	soundEffect = loadSound("static/audio/sound effect.mp3");
+
+	background(200);
+	textAlign(CENTER);
+	buttons = makeButtonsFromModes(modes);
+}
+
+function draw() {
+	if (timerRunning) pomodoro(currentWorkTime, currentRestTime);
+}
+
+function resetProgram() {
+	//executes once to reset program
+	buttons.forEach((b) => b.remove());
```

Comment:
I do see this call repeated in a few places. It's probably wise to extract this into its own function.